platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/Hash.java

```diff
@@ -262,7 +245,7 @@ public int hashCode() {
      * {@inheritDoc}
      */
     @Override
-    public int compareTo(final Hash that) {
+    public int compareTo(@NonNull final Hash that) {
```

Comment:
`@NonNull` not necessary as tested explicitly at L253-255.

Perhaps this is an IntelliJ catch-22?  It claims you _need_ the `@NonNull` annotation because the overridden method in interface `java.lang.Comparable` is annotated `@NonNull` - but that annotation is IntelliJ's own doing!  The _real_ method isn't annotated that way! See https://github.com/openjdk/jdk/blob/0bb5ae645165b97527ecccf02308df6072c363d8/src/java.base/share/classes/java/lang/Comparable.java#L141 - and it is in fact declared to throw `NullPointerException` when the argument is `null` - https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Comparable.html#compareTo(T)